polar-core/src/vm.rs

```diff
@@ -1171,50 +1138,29 @@ impl PolarVirtualMachine {
                 ));
 
                 // Construct field constraints.
-                fields.fields.iter().rev().for_each(|(f, v)| {
+                let field_constraints = fields.fields.iter().rev().map(|(f, v)| {
+                    let v = self.deref(v);
                     let field = right.clone_with_value(value!(f.0.as_ref()));
                     let left = left.clone_with_value(value!(op!(Dot, left.clone(), field)));
-                    let mut unify = op!(Unify, left.clone(), v.clone());
-                    match v.value() {
-                        Value::Variable(v) | Value::RestVariable(v) => {
-                            match self.variable_state(v) {
-                                VariableState::Cycle(c) => {
-                                    unify = cycle_constraints(c)
-                                        .clone_with_new_constraint(unify.into_term());
-                                }
-                                VariableState::Bound(b) => {
-                                    unify = op!(Unify, left, b);
-                                }
-                                VariableState::Partial(e) => {
-                                    unify = e.clone_with_new_constraint(unify.into_term());
-                                }
-                                VariableState::Unbound => (),
-                            }
-                        }
-                        _ => (),
-                    }
-                    operation = operation.clone_with_new_constraint(term!(unify));
+                    op!(Unify, left, v)
```

Comment:
Should this be `Isa`?

Reply:
Yeah, it should

Reply:
Looks like the non constraint code uses Isa, not Unify.